src/site/topic8.rst

```diff
@@ -135,6 +135,31 @@ Size and isEmpty
     * Can you think of another way to check if it's empty?
 
 
+toString
+^^^^^^^^
+
+.. code-block:: java
+    :linenos:
+
+        @Override
+        public String toString() {
+            StringBuilder builder = new StringBuilder();
+            builder.append(", ");
+            Node<T> currentNode = top;
+            while (currentNode != null) {
+                builder.insert(0, currentNode.getData());
+                builder.insert(0, ", ");
+                currentNode = currentNode.getNext();
+            }
+            builder.delete(0, 2);
+            builder.append("<-- Top\n");
+            return builder.toString();
+        }
+
+* It's a little ugly here
```

Comment:
It is indeed. Only thing I can think of off the top of my head is the reverse the list; so going from bottom up to generate the string. A single annoying for singly linked, so eh. It's more of an implementation detail than the point of the lecture, so I think it's fine for them being able to play with it at least.

Reply:
Once we hit iterators, this should clean up nicely.